ChaosMod/Effects/db/Player/PlayerBigBang.cpp

```diff
@@ -0,0 +1,92 @@
+#include <stdafx.h>
+
+//Effect by ProfessorBiddle, Based on code from Pongo and Last0xygen
+
+static void OnStart()
+{
+	std::vector<Entity> entities;
+	int maxEntities = 20;
+
+	for (auto ped : GetAllPeds())
+	{
+		if (maxEntities == 10)
+		{
+			break;
+		}
+
+		maxEntities--;
+
+		if (!IS_PED_A_PLAYER(ped) && !IS_ENTITY_A_MISSION_ENTITY(ped))
+		{
+			SET_ENTITY_AS_MISSION_ENTITY(ped, true, true);
+			entities.push_back(ped);
+		}
+	}
+
+	auto playerPed = PLAYER_PED_ID();
+
+	for (auto veh : GetAllVehs())
+	{
+		if (maxEntities == 0)
+		{
+			break;
+		}
+
+		maxEntities--;
+
+		if ((!IS_PED_IN_ANY_VEHICLE(playerPed, false) || veh != GET_VEHICLE_PED_IS_IN(playerPed, false)) && !IS_ENTITY_A_MISSION_ENTITY(veh))
```

Comment:
the vehicle could be checked outside the for-loop so it doesnt have to be checked all the time

Reply:
fixed